src/IngredientList.js

```diff
@@ -0,0 +1,37 @@
+import React from 'react'
+import Ingredient from './Ingredient'
+import NewIngredientForm from './NewIngredientForm'
+
+const IngredientList = ({
+    recipe = { ingredients: [] },
+    isEditing,
+    addIngredient,
+    removeIngredient,
+}) => {
+    const add = (newIngredient) => {
+        addIngredient(newIngredient)
+    }
+
+    const remove = ({ name }) => {
+        removeIngredient(name)
+    }
+
+    return (
+        <div>
+            <h3>Ingredients</h3>
+            {isEditing && <NewIngredientForm onNewIngredient={add} />}
+
+            {recipe.ingredients &&
+                recipe.ingredients.map((ingredient) => (
```

Comment:
another way to achieve the same is using the optional chaining operator
```suggestion
            {recipe?.ingredients.map((ingredient) => (
```
and you can also save the default value on the component props